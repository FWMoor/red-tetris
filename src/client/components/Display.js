import React from 'react';
import { Button, Typography } from '@material-ui/core';

import { StyledDisplay } from './styles/StyledDisplay';

const Display = ({ gameOver, text }) => {
	return (
		<StyledDisplay gameOver={gameOver}>
			<Typography>{text}</Typography>
		</StyledDisplay>
	);
};

export default Display;
