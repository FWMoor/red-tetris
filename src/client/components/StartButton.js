import React from 'react';
import { Button, Typography } from '@material-ui/core';

import { StyledStartButton } from './styles/StyledStartButton';

const StartButton = ({ callback }) => (
	<Button
		variant="contained"
		color="warning"
		id="startButton"
		onClick={callback}
		fullWidth="true"
	>
		{/* <StyledStartButton id="startButton" onClick={callback}> */}
		<Typography>START GAME</Typography>
		{/* </StyledStartButton> */}
	</Button>
);

export default StartButton;
