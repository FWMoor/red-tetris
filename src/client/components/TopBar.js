import React from 'react';
import {
	Toolbar,
	Typography,
	AppBar,
	Button,
	IconButton,
} from '@material-ui/core';
import { MenuIcon } from '@material-ui/icons/Menu';

const TopBar = () => (
	<AppBar position="static">
		<Toolbar color="secondary">
			<IconButton edge="start" color="inherit" aria-label="menu">
				<MenuIcon />
			</IconButton>
			<Button color="inherit">Login</Button>
			<Typography>Red Tetris</Typography>
		</Toolbar>
	</AppBar>
);

export default TopBar;
