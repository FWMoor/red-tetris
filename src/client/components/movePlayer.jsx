import { checkCollision } from "../helpers/gameHelpers";

export const movePlayer = (dir, updatePlayerPos, player, stage) => {
	if (!checkCollision(player, stage, { x: dir, y: 0 })) {
		updatePlayerPos({ x: dir, y: 0 });
	}
};

export const dropPlayer = (setDropTime, drop, rows, level, player, stage, setLevel, updatePlayerPos, setGameOver) => {
	console.log('Interval off');
	setDropTime(null);
	drop(rows, level, player, stage, setLevel, setDropTime, updatePlayerPos, setGameOver);
};

export const drop = (rows, level, player, stage, setLevel, setDropTime, updatePlayerPos, setGameOver) => {
	// Increase level when player has cleared 10 rows
	if (rows > (level + 1) * 10) {
		setLevel((prev) => prev + 1);
		// Also increase the speed
		setDropTime(1000 / (level + 1) + 200);
	}

	if (!checkCollision(player, stage, { x: 0, y: 1 })) {
		updatePlayerPos({ x: 0, y: 1, collided: false });
	} else {
		// Game Over
		if (player.pos.y < 1) {
			console.log('GAME OVER!');
			setGameOver(true);
			setDropTime(null);
		}
		updatePlayerPos({ x: 0, y: 0, collided: true });
	}
};

export const playerRotation = (stage, dir, playerRotateFunc) => {
	playerRotateFunc(stage, dir);
}