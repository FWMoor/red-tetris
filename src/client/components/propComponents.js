
import {StyledDisplay} from './styles/StyledDisplay';
import  {StyledCell} from './styles/StyledCell';


export{
	StyledDisplay,
	StyledCell,
	// Cell
	// StyledStage,
	// StyledStartButton,
	// StyledTetris
}