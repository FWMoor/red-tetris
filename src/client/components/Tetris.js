import React, { useState } from 'react';

import { createStage, checkCollision } from '../helpers/gameHelpers';
import userSocket from "../helpers/socket";

// Styled Component
import { StyledTetrisWrapper, StyledTetris } from './styles/StyledTetris';

// Custom Hooks
import { useInterval } from '../hooks/useInterval';
import { usePlayer } from '../hooks/usePlayer';
import { useStage } from '../hooks/useStage';
import { useGameStatus } from '../hooks/useGameStatus';
import { movePlayer, dropPlayer, drop, playerRotation } from './movePlayer';
// Components
import Stage from './Stage';
import Display from './Display';
import StartButton from './StartButton';

let mainSocket = null

const Tetris = () => {
	// console.log(createStage());

	const [dropTime, setDropTime] = useState(null);
	const [gameOver, setGameOver] = useState(false);

	const { player, updatePlayerPos, resetPlayer, playerRotate } = usePlayer();
	const { stage, setStage, rowsCleared, addRow } = useStage(player, resetPlayer, mainSocket);
	const { score, setScore, rows, setRows, level, setLevel } = useGameStatus(
		rowsCleared
	);

	const startGame = () => {
		// Reset everything
		setStage(createStage());
		setDropTime(1000);
		resetPlayer();
		setGameOver(false);
		setScore(0);
		setRows(0);
		setLevel(1);
	};

	const keyUp = ({ keyCode }) => {
		if (!gameOver) {
			if (keyCode === 40) {
				setDropTime(1000 / (level + 1) + 200);
				console.log('Interval on: 1000');
			}
		}
	};

	const move = ({ keyCode }) => {
		if (!gameOver) {
			// 37 = left arrow, -1 on x axis
			// 39 = right arrow, +1 on x axis
			// 40 = down arrow
			// 38 = up arrow, rotate
			if (keyCode === 37) {
				movePlayer(-1, updatePlayerPos, player, stage);
			} else if (keyCode === 39) {
				movePlayer(1, updatePlayerPos, player, stage);
			} else if (keyCode === 40) {
				dropPlayer(setDropTime, drop, rows, level, player, stage, setLevel, updatePlayerPos, setGameOver);
			} else if (keyCode === 38) {
				playerRotation(stage, 1, playerRotate);
			}
		}
	};

	useInterval(async () => {
		mainSocket = await userSocket();
		mainSocket.on('idk', () => {
			addRow(stage)
			updatePlayerPos({ x: 0, y: 0, collided: false })
		})
		drop(rows, level, player, stage, setLevel, setDropTime, updatePlayerPos, setGameOver);
	}, dropTime);

	return (
		<StyledTetrisWrapper
			role="button"
			tabIndex="0"
			onKeyDown={(e) => move(e)}
			onKeyUp={keyUp}
		>
			<StyledTetris>
				<Stage stage={stage} />
				<aside>
					{gameOver ? (
						<Display id="gameOverDisplay" gameOver={gameOver} text="Game Over" />
					) : (
							<div id="test">
								<Display id="scoreDisplay" text={`Score: ${score}`} />
								<Display id="rowDisplay" text={`Rows: ${rows}`} />
								<Display id="levelDisplay" text={`Level: ${level}`} />
							</div>
						)}
					<StartButton callback={startGame} />
				</aside>
			</StyledTetris>
		</StyledTetrisWrapper>
	);
};

export default Tetris;
