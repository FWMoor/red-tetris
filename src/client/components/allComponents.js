import {Test} from './test';

// Components
import Cell from './Cell';
import Display from './Display';
import Stage from './Stage';
import StartButton from './StartButton';
import Tetris from './Tetris';

// Component Styles
import StyledCell from './styles/StyledCell';
import StyledDisplay from './styles/StyledDisplay';
import StyledStage from './styles/StyledStage';
import StyledStartButton from './styles/StyledStartButton';
import StyledTetris from './styles/StyledTetris';

export {
	// Test,
	Cell,
	Display,
	Stage,
	StartButton,
	Tetris,
	StyledCell,
	StyledDisplay,
	StyledStage,
	StyledStartButton,
	StyledTetris,
};
