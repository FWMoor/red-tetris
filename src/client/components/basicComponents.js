
// Components
import Display from './Display';
import Stage from './Stage';
import StartButton from './StartButton';
import Tetris from './Tetris';
import TopBar  from './TopBar';

export {
	Display,
	// Stage,
	StartButton,
	Tetris, 
	TopBar
};
