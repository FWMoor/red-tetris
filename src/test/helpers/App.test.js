import React from 'react';
import ReactDOM from 'react-dom';
import App from '../../client/containers/app';
import * as redux from 'react-redux'
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import * as components from '../../client/components/allComponents'

Enzyme.configure({ adapter: new Adapter() })
it('renders correctly enzyme', () => {
	const spy = jest.spyOn(redux, 'useSelector')
	spy.mockReturnValue('Soon, will be here a fantastic Tetris ...')
	const wrapper = shallow(<App />)
	expect(toJson(wrapper)).toMatchSnapshot();
  });
describe("tests components render properly",() =>{
	Object.keys(components).map(component =>{
		const Component = component
		it(`tests ${component}`, () =>{
			const wrapper = shallow(<Component />)
			expect(toJson(wrapper)).toMatchSnapshot();
		})
	})
})