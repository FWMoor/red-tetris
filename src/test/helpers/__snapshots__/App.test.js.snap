// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly enzyme 1`] = `
<span>
  Soon, will be here a fantastic Tetris ...
</span>
`;

exports[`tests components render properly tests Test 1`] = `<Test />`;
